src/controller/index.ts

```diff
@@ -304,37 +324,86 @@ export default class SimulariumController {
         }
     }
 
-    public gotoTime(time: number): void {
-        // If in the middle of changing files, ignore any gotoTime requests
+    private clampFrameNumber(frame: number): number {
+        return Math.max(0, Math.min(frame, this.visData.totalSteps - 1));
+    }
+
+    private getFrameAtTime(time: number): number {
+        const frameNumber = Math.round(time / this.visData.timeStepSize);
+        const clampedFrame = this.clampFrameNumber(frameNumber);
+        return clampedFrame;
+    }
+
+    public movePlaybackFrame(frameNumber: number): void {
+        if (this.streaming) {
```

Comment:
if (isPlaying || isPrefetching) this.pauseStreaming. 

if (isPlaying) pause 
if (isPrefetching) pausePrefetching